Modules/Shop/Http/Services/ShopService.php

```diff
@@ -629,4 +771,238 @@ public function getProduct(int $productId): array
             return ['res' => false, 'msg' => "Product not found !", 'data' => ''];
         }
     }
+
+    /**
+     * Get a listing of new Brands
+     *
+     * @return array
+     */
+    public function getNewBrands(): array
+    {
+        $data = [];
+        $brandUsers = User::where('role', 'brand')->where('created_at', '>', now()->subDays(45)->endOfDay())->get();
+
+        if ($brandUsers) {
+            foreach ($brandUsers as $brandUser) {
+                $brand = Brand::where('user_id', $brandUser['id'])->where('go_live', '2')->first();
+                if ($brand) {
+                    $data[] = array(
+                        'brand_key' => $brand->bazaar_direct_link,
+                        'brand_id' => $brand->id,
+                        'brand_name' => $brand->brand_name,
+                        'brand_logo' => $brand->logo_image != '' ? asset('public') . '/' . $brand->logo_image : asset('public/img/logo-image.png'),
+                    );
+                }
+
+            }
+        }
+
+        return ['res' => true, 'msg' => "", 'data' => $data];
+    }
+
+    /**
+     * Search brands or products
+     *
+     * @param Object $request
+     * @return array
+     */
+    public function search(object $request): array
+    {
+        $product = [];
+        $getBrand = [];
+
+        $results = Product::leftJoin('categories as main', 'products.main_category', '=', 'main.id')
+            ->leftJoin('categories as sub', 'products.category', '=', 'sub.id')
+            ->leftJoin('categories as child', 'products.sub_category', '=', 'child.id')
+            ->where('products.status', 'publish')
+            ->where('products.name', 'like', '%'.$request->search.'%')
+            ->orWhere('main.title', 'like', '%'.$request->search.'%' )
+            ->orWhere('sub.title', 'like', '%'.$request->search.'%' )
+            ->orWhere('child.title', 'like', '%'.$request->search.'%' )
+            ->get();;
```

Comment:
@arijit1111 If you need only one column as `name` you can use the `select` method to select specific columns.

Reply:
fixed